src/System.Net.Mail/src/System/Net/Mail/NTAuthenticationStub.cs

```diff
@@ -0,0 +1,96 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Reflection;
+using System.Net.Security;
+using System.Security.Authentication.ExtendedProtection;
+using System.Diagnostics;
+
+namespace System.Net
+{
+    internal class NTAuthentication
+    {
+        private static readonly Type s_type = typeof(NegotiateStream).Assembly.GetType("System.Net.NTAuthentication");
+        private const BindingFlags NonPublicInstance = BindingFlags.NonPublic | BindingFlags.Instance;
+
+        private readonly ConstructorInfo _ntAuthentication;
+        private readonly MethodInfo _getOutgoingBlogString;
+        private readonly MethodInfo _getOutgoingBlobBytes;
+        private readonly MethodInfo _verifySignature;
+        private readonly MethodInfo _makeSignature;
+        private readonly MethodInfo _closeContext;
+        private readonly MethodInfo _isCompleted;
+        private object _instance;
+
+        internal NTAuthentication(bool isServer, string package, NetworkCredential credential, string spn, ContextFlags requestedContextFlags, ChannelBinding channelBinding)
+        {
+            ConstructorInfo[] ctors = s_type.GetConstructors(NonPublicInstance & ~BindingFlags.Default);
+            Debug.Assert(ctors.Length == 1);
+
+            _ntAuthentication = ctors[0];
+            _getOutgoingBlogString = s_type.GetMethod("GetOutgoingBlob", NonPublicInstance, null, new Type[] { typeof(string)}, null);
+            _getOutgoingBlobBytes = s_type.GetMethod("GetOutgoingBlob", NonPublicInstance, null, new Type[] { typeof(byte[]), typeof(bool) }, null);
+            _verifySignature = s_type.GetMethod("VerifySignature", NonPublicInstance);
+            _makeSignature = s_type.GetMethod("MakeSignature", NonPublicInstance);
+            _closeContext = s_type.GetMethod("CloseContext", NonPublicInstance);
+            _isCompleted = s_type.GetProperty("IsCompleted", NonPublicInstance).GetMethod;
+
+            _instance = _ntAuthentication.Invoke(new object[] { isServer, package, credential, spn, requestedContextFlags, channelBinding });
+        }
+
+        internal string GetOutgoingBlob(string challenge)
+        {
+            return (string)_getOutgoingBlogString.Invoke(_instance, new object[] { challenge });
```

Comment:
You'd have to measure, but it might be faster to lazily allocate (and cache in an instance field) a delegate via `MethodInfo.CreateDelegate` and then call the delegate instead of calling `MethodInfo.Invoke`.